docs/src/modules/components/overview/charts/mainDemo/DownloadDemo.tsx

```diff
@@ -0,0 +1,154 @@
+
+  for (const version of versions) {
+    if (version in item) {
+      item[`${version}_percent`] = (100 * (item[version] ?? 0)) / sum;
+    } else {
+      item[version] = null;
+      item[`${version}_percent`] = null;
+    }
+  }
+  return item;
+};
+
+const versions: Record<string, Versions[]> = {
+  '@mui/material': ['5', '6', '7'],
+  '@mui/x-data-grid': ['5', '6', '7', '8'],
+  '@mui/x-date-pickers': ['5', '6', '7', '8'],
+  '@mui/x-charts': ['6', '7', '8'],
+};
+
+const packages = {
+  '@mui/material': dataMaterial.map((item) =>
+    addRelativeValue(versions['@mui/material'])({ ...item, date: new Date(item.date) }),
+  ),
+  '@mui/x-data-grid': dataDataGrid.map((item) =>
+    addRelativeValue(versions['@mui/x-data-grid'])({ ...item, date: new Date(item.date) }),
+  ),
+  '@mui/x-date-pickers': dataPickers.map((item) =>
+    addRelativeValue(versions['@mui/x-date-pickers'])({ ...item, date: new Date(item.date) }),
+  ),
+  '@mui/x-charts': dataCharts.map((item) =>
+    addRelativeValue(versions['@mui/x-charts'])({ ...item, date: new Date(item.date) }),
+  ),
+} as const;
+
+const IntlAbsolute = new Intl.NumberFormat('en-US');
+const valueFormatter = (value: number | null) => {
+  if (value !== null) {
+    return IntlAbsolute.format(value);
+  }
+  return value;
+};
+
+const dateFormatter = new Intl.DateTimeFormat('en-US', {
+  year: 'numeric',
+  month: 'short',
+  day: 'numeric',
+}).format;
+
+const IntlPercent = new Intl.NumberFormat('en-US', { style: 'percent' });
+const percentValueFormatter = (value: number | null) => {
+  if (value !== null) {
+    return IntlPercent.format(value / 100);
+  }
+  return value;
+};
+
+export default function DownloadDemo() {
+  const [selectedPackage, setSelectedPackage] =
+    React.useState<keyof typeof packages>('@mui/material');
+
+  const [selectedFormat, setSelectedFormat] = React.useState<'absolute' | 'relative'>('absolute');
+
+  return (
+    <Stack spacing={1} direction="column" sx={{ height: '100%', minHeight: 0 }}>
+      <Stack direction="row" justifyContent="space-between" width="100%">
+        <Select
+          size="small"
+          sx={{ width: 200 }}
+          value={selectedPackage}
+          onChange={(event) => setSelectedPackage(event.target.value as keyof typeof packages)}
+        >
+          {Object.keys(packages).map((pkg) => (
+            <MenuItem key={pkg} value={pkg}>
+              {pkg}
+            </MenuItem>
+          ))}
+        </Select>
+
+        <Select
+          size="small"
+          sx={{ width: 200 }}
+          value={selectedFormat}
+          onChange={(event) => setSelectedFormat(event.target.value as 'absolute' | 'relative')}
+        >
+          <MenuItem value="absolute">Absolute</MenuItem>
+          <MenuItem value="relative">Relative</MenuItem>
+        </Select>
+      </Stack>
+
+      <div style={{ flex: 1, minHeight: 0 }}>
+        <LineChartPro
```

Comment:
![image](https://github.com/user-attachments/assets/3bfefe30-d8b9-4160-844f-b782f2132254)

Should we format the Y axis using the M suffix instead? "9,000,000" -> "9M".

Also, the X axis ticks should probably be formatted using a short month and maybe we should also include the year? Otherwise it isn't clear what year is being displayed (see image above)